### PR TITLE
[MRG+1] Add Allen RSN templates fetcher v0.2

### PR DESCRIPTION
**Pull request by alexsavio**

### CODE REVIEW

File: nilearn/datasets/tests/test_atlas.py [hunk 1]
```diff
@@ -398,3 +398,23 @@ def test_fetch_coords_dosenbach_2010():
...
+                                               verbose=0)
+    keys = ("all_unthresh_tmaps",
+            "rsn_unthresh_tmaps",
+            "aggregate_ic_comps")
+
+    filenames = ["ALL_HC_unthresholded_tmaps.nii",
+                 "RSN_HC_unthresholded_tmaps.nii",
+                 "rest_hcp_agg__component_ica_.nii"]
+
+    assert_equal(len(tst.mock_url_request.urls), 1)
```
Comment (AlexandreAbraham) [inline nilearn/datasets/tests/test_atlas.py]: This should be 3 and not 1 given that 3 files are downloaded.

**Comment by KamalakerDadi** [comment]
Can you add in ```nilearn/datasets/__init__.py``` ?

**Comment by KamalakerDadi** [comment]
Can you add in ```doc/whats_new.rst``` about this new fetcher ?

**Comment by KamalakerDadi** [comment]
Can you please also refer your fetcher in doc/modules/reference.rst ?

File: nilearn/datasets/atlas.py [hunk 2]
```diff
@@ -782,3 +782,83 @@ def fetch_coords_dosenbach_2010(ordered_regions=True):
                   networks=out_csv['network'], description=fdescr)
 
     return Bunch(**params)
+
+
+def fetch_allen_rsn_tmap_75_2011(data_dir=None, url=None, resume=True, verbose=1):
```
Comment (KamalakerDadi) [inline nilearn/datasets/atlas.py]: May be this sounds good ```fetch_atlas_allen_2011``` ?

**Comment by KamalakerDadi** [comment]
Some minor comments from my side.

File: nilearn/datasets/atlas.py (hunk 2 above)
Comment (KamalakerDadi) [inline nilearn/datasets/atlas.py]: >Would you call this an atlas? 

Group of RSNs can be called as an atlas.

>If you want to keep the atlas

That is how rest of the functions are kept as. I have no strong opinions on the names. My intention is just to keep them easy to call and consistent with other functionalities.

Comment (KamalakerDadi) [inline nilearn/datasets/atlas.py]: @GaelVaroquaux Any suggestions on the name ?

Comment (alexsavio) [inline nilearn/datasets/atlas.py]: I just saw that the fetch_atlas_smith_2009 is also for a RSN template. So your suggestion sounds good.

**Comment by alexsavio** [comment]
There is an error coming from `fetch_atlas_smith_2009`, do you want me to fix that for this PR?

**Comment by KamalakerDadi** [comment]
>do you want me to fix that for this PR?

We are discussing the fix in issue #1307. After we agree upon one fix. I would suggest to fix in another PR rather than here.

**Comment by KamalakerDadi** [comment]
Failures are now fixed in master. Can you please rebase ?

It seems like you have added whats_new changes in 0.2.6. 0.2.6 is already released and your additions will reflect in upcoming release. Therefore, can you add a new release section with 0.2.6.1 and then include your new function details there ? Thanks.

**Comment by alexsavio** [comment]
I can't do a proper rebase.

**Comment by KamalakerDadi** [comment]
>I can't do a proper rebase.

 Me neither. I can guide you but I am afraid that it gets more complicated. Do you have any other alternative ?

**Comment by alexsavio** [comment]
@KamalakerDadi my alternative is always pull everything, merge the changes manually and do another PR.

**Comment by GaelVaroquaux** [comment]
>     I can't do a proper rebase.

> Me neither.

OK, don't bother with rebase. Merge is OK.

**Comment by GaelVaroquaux** [comment]
Tests are all failing here, and it doesn't look spurious to me. @alexsavio : could you please have a look?

**Comment by alexsavio** [comment]
I will, soon.

**Comment by alexsavio** [comment]
For the sake of a clean history I am doing a different PR.

**Pull request closed without merging**